templates/docs/patterns/newsletter-signup/index.md

```diff
@@ -0,0 +1,316 @@
+---
+wrapper_template: '_layouts/docs.html'
+context:
+  title: Newsletter signup | Patterns
+---
+
+{% from "docs/macros/patterns/wip-notice.jinja" import pattern_wip_notice %}
+
+{{- pattern_wip_notice() }}
+
+Newsletter signup is a form which allows users to subscribe to a newsletter for updates.
+
+- [25-75](#25-75)
+- [50-50](#50-50)
+- [col-2](#col-2)
+- [col-4](#col-4)
+
+The Newsletter signup pattern is composed of the following elements:
+
+| Element                                 | Description                                       |
+| --------------------------------------- | ------------------------------------------------- |
+| Title (**required**)                    | Title text (to be placed in `h2` or `h3` heading) |
+| - `h2` for `50-50` and `25-75` variants |
+| - `h3` for `col-2` and `col-4` variants |
+| Description (**required**)              | Description text (one or more paragraphs)         |
+| Label + input (**required**)            | Single-line input field with label                |
+| Label + checkbox (**required**)         | Checkbox with a label                             |
+| Button (**required**)                   | Single default button                             |
+
+## 25-75
+
+This is the default variant. It consists of a 25-75 split, with title on the left, and description(s) with form on the right.
+
+<div class="embedded-example"><a href="/docs/examples/patterns/newsletter-signup/25-75" class="js-example" data-lang="jinja">
+View example of the 25-75 newsletter signup
+</a></div>
+
+## 50-50
+
+THis variant consists of a 50-50 split, with title on the left, and description(s) with form on the right.
+
+<div class="embedded-example"><a href="/docs/examples/patterns/newsletter-signup/50-50" class="js-example" data-lang="jinja">
+View example of the 50-50 newsletter signup
+</a></div>
+
+## 2-col
+
+This variant is used within a grid, and takes up 2 columns on large/medium and 4 columns on small dimensions.
+
+<div class="embedded-example"><a href="/docs/examples/patterns/newsletter-signup/2-col" class="js-example" data-lang="jinja">
+View example of the 2-col newsletter signup
+</a></div>
+
+## 4-col
```

Comment:
A reader who is less familiar with the Vanilla grid system may not understand that "2-col" refers to how many columns are used **internally** by the newsletter form itself. In the example rendered here, you say "2-col" but visually we see four columns on the page.

It's important that the variants here map very closely with the [figma](https://www.figma.com/design/1brHCsyeTE6FU7hfJd6ao2/%F0%9F%9A%A7---WIP--Sites---Core-component-library?node-id=567-5903&p=f&m=dev) like you have already done, but I would recommend adding a note about the 2-col and 4-col variants that the **form itself** takes up **n** columns, and the other **colMax - n** columns are populated by the caller with the `col_` slots.